pkg/controller/controller.go

```diff
@@ -115,22 +127,22 @@ func (c *Controller) HandlePopup() http.Handler {
 
 // handlePopupGet handles the initial page load.
 func (c *Controller) handlePopupGet(w http.ResponseWriter, r *http.Request) {
-	formDetails, err := getFormDetails(r)
+	formDetails, err := c.getFormDetails(r)
 	if err != nil {
 		c.renderBadRequest(w, err.Error())
 		return
 	}
 
 	// set some defaults for the form
-	formDetails.Category = categories()[0]
+	formDetails.Category = c.categories()[0]
```

Comment:
I realize this is the existing code, but this can panic if the categories is empty (index out of bounds)

Reply:
"explanation" is hard-coded below, as a result, it won't through exception. Seems the logic is "explanation" should always be one of the category to allow user to input free style text.

https://github.com/abcxyz/jvs/blob/main/pkg/controller/controller.go#L338

Reply:
I fixed this in https://github.com/abcxyz/jvs/pull/303; we shouldn't assume the value will always be hardcoded (or have a specific length)

Reply:
Thanks a lot for fixing it. Merging #303 now.